src/Stratis.FederatedPeg.Features.FederationGateway/Controllers/FederationGatewayController.cs

```diff
@@ -120,41 +126,14 @@ public IActionResult ResyncMaturedBlockDeposits([FromBody] MaturedBlockModel blo
 
             try
             {
-                ChainedHeader chainedHeader = this.chain.GetBlock(blockHashHeight.BlockHash);
-
-                if (chainedHeader == null)
-                {
-                    return ErrorHelpers.BuildErrorResponse(HttpStatusCode.BadRequest, $"Block with hash {blockHashHeight.BlockHash} was not found on the block chain.", string.Empty);
-                }
-
-                int currentHeight = chainedHeader.Height;
-                int matureHeight = (this.chain.Tip.Height - (int)this.depositExtractor.MinimumDepositConfirmations);
-
-                if (currentHeight > matureHeight)
-                {
-                    return ErrorHelpers.BuildErrorResponse(HttpStatusCode.BadRequest,
-                        $"Block height {blockHashHeight.BlockHeight} submitted is not mature enough. Blocks less than a height of {matureHeight} can be processed.", string.Empty);
-                }
-
-                while (currentHeight < matureHeight)
-                {
-                    IMaturedBlockDeposits maturedBlockDeposits =
-                        this.depositExtractor.ExtractMaturedBlockDeposits(chainedHeader);
+                List<IMaturedBlockDeposits> deposits = this.maturedBlocksProvider.GetMaturedDepositsAsync(
+                    blockRequest.BlockHeight, blockRequest.MaxBlocksToSend).GetAwaiter().GetResult();

```

Comment:
Could await this and make the method return async Task<IActionResult>.

Reply:
Done.